sdk/eventhub/Azure.Messaging.EventHubs/README.md

```diff
@@ -1,16 +1,163 @@
+
+- A **partition** is an ordered sequence of events that is held in an Event Hub. Partitions are a means of data organization associated with the parallelism required by event consumers.  Azure Event Hubs provides message streaming through a partitioned consumer pattern in which each consumer only reads a specific subset, or partition, of the message stream. As newer events arrive, they are added to the end of this sequence. The number of partitions is specified at the time an Event Hub is created and cannot be changed.
+
+- A **consumer group** is a view of an entire Event Hub. Consumer groups enable multiple consuming applications to each have a separate view of the event stream, and to read the stream independently at their own pace and from their own position.  There can be at most 5 concurrent readers on a partition per consumer group; however it is recommended that there is only one active consumer for a given partition and consumer group pairing. Each active reader receives all of the events from its partition; If there are multiple readers on the same partition, then they will receive duplicate events. 
+
+For more concepts and deeper discussion, see: [Event Hubs Features](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-features)
+
+## Examples
+
+### Inspect an Event Hub
+
+Many Event Hub operations take place within the scope of a specific partition.  Because partitions are owned by the Event Hub, their names are assigned at the time of creation.  To understand what partitions are available, you query the Event Hub using the client.
+
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+await using (var client = new EventHubClient(connectionString, eventHubName))
+{
+    string[] partitionIds = await client.GetPartitionIdsAsync();
+}
+```
+
+### Consume events from an Event Hub
+
+In order to consume events, you'll need to create an `EventHubConsumer` for a specific partition and consumer group combination.  When an Event Hub is created, it starts with a default consumer group that can be used to get started.  A consumer also needs to specify where in the event stream to begin receiving events; in our example, we will focus on reading new events as they are published.
+
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+await using (var client = new EventHubClient(connectionString, eventHubName))
+{
+    string firstPartition = (await client.GetPartitionIdsAsync()).First();
+    string consumerGroup = EventHubConsumer.DefaultConsumerGroup;
+    EventPosition startingPosition = EventPosition.Latest;
+    
+    await (using EventHubConsumer consumer = client.CreateConsumer(consumerGroup, firstPartition, startingPosition))
```

Comment:
It's intentional.   It's the `IAsyncDisposable` syntax that is forthcoming in C# 8, and which we're using in preview form via NuGet package.   The alternative is to explicitly layout the `try/finally` block to close these properly which I am concerned adds too much clutter and length to the snippets, taking away from the meaningful part.   That would have things looking similar to:

```csharp
var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
var eventHubName = "<< NAME OF THE EVENT HUB >>";
var client = new EventHubClient(connectionString, eventHubName);

EventHubConsumer consumer;

try
{
    string firstPartition = (await client.GetPartitionIdsAsync()).First();
    string consumerGroup = EventHubConsumer.DefaultConsumerGroup;
    EventPosition startingPosition = EventPosition.Latest;

    consumer = client.CreateConsumer(consumerGroup, firstPartition, startingPosition));

    int maximumEventBatchSize = 25;
    IEnumerable<EventData> eventBatch = await consumer.Receive(maximumEventBatchSize);
 
        // At this point, the eventBatch may have no events or may have as many as the maximum size requested, 
        // depending on how many events were available in the partition.
}
finally
{
    await Task.WhenAll(client.CloseAsync(), (consumer?.CloseAsync() ?? Task.CompletedTask));
}
```

Though the `await using` syntax is newer and may not be as well known, I prefer the brevity in it so that the more important parts that I'm trying to illustrate come to the forefront.

Reply:
... and from offline discussion, I totally missed the actual question.    Yes, the parenthesis should be after the `using` ... nice catch!